infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/EndpointMetadata.java

```diff
@@ -378,6 +388,24 @@ private void writeParameters(
     }
   }
 
+  private void writeHeaders(
+      final JsonGenerator gen,
+      final Map<String, StringValueTypeDefinition<?>> fields,
+      final boolean isMandatoryField)
+      throws IOException {
+    for (Map.Entry<String, StringValueTypeDefinition<?>> entry : fields.entrySet()) {
+      gen.writeStartObject();
+      gen.writeObjectField("name", entry.getKey());
+      if (isMandatoryField) {
+        gen.writeObjectField("required", true);
+      }
+      gen.writeObjectField("in", "header");
+      gen.writeFieldName("schema");
+      entry.getValue().serializeOpenApiTypeOrReference(gen);
+      gen.writeEndObject();
+    }
+  }
```

Comment:
It looks like this method is not necessary as we could use `writeParameters(..)` passing `paramUsedIn="header"`. Maybe check if we could reuse that method.